File: frontend/src/app/types/Order.ts
import { Product } from "./Product";

export interface Order {
  id: number;
  email: string;
  address: string;
  postalCode: string;
  state: string;
  totalPrice: number;
  orderDate: string;
}

export interface OrderResponseDto {
  orders: Order[];
}

export interface OrderRequestDto {
  email: string;
  address: string;
  postalCode: string;
  totalPrice: number;
  products: {
    productId: number;
    quantity: number;
  }[];
}

export interface OrderDetailResponseDto {
  products: {
    name: string;
    price: number;
    quantity: number;
    imgPath: string;
  }[];
}

File: backend/src/main/java/com/ll/backend/domain/order/entity/Order.java
package com.ll.backend.domain.order.entity;

import com.ll.backend.domain.order.OrderStatus;
import com.ll.backend.domain.orderdetail.entity.OrderDetail;
import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Getter
@Setter
@Table(name = "ORDERS")
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "order_id")
    private Integer id;

    @Column(length = 100)
    @Email
    private String email;

    @Column(length = 100)
    private String address;

    @Column(length = 20)
    private String postalCode;

    @Column(length = 50)
    @Enumerated(EnumType.STRING)
    private OrderStatus state;

    private int totalPrice;

    private LocalDateTime orderDate;

    @OneToMany(mappedBy = "order", fetch = FetchType.LAZY)
    private List<OrderDetail> orderDetails;

    public Order(String email, String address, String postalCode, OrderStatus state, int totalPrice) {
        this.email = email;
        this.address = address;
        this.postalCode = postalCode;
        this.state = state;
        this.totalPrice = totalPrice;
        this.orderDate = LocalDateTime.now();
    }


    public Order() {

    }
}
